src/components/mixins/UtilityMixin.vue

```diff
@@ -64,5 +64,19 @@ export default class UtilityMixin extends Vue {
 
         this.tsRefreshFailed = false;
     }
+
+    /**
+     * Return true if the main level route uses navigation component.
+     *
+     * Main route refers to the first level of routes definition, e.g.
+     * "manager", to differentiate it from child routes, e.g.
+     * "manager.installed". In practice this means that if the main
+     * route uses a navigation component, it's expected that the child
+     * routes use one as well.
+     */
+    routeHasNavbar() {
```

Comment:
Not entirely sure how this works?

How does this determine whether SettingsView/Help have a navbar as they don't have sub routes?
How is it hidden on the Profile selection screen for example?

Reply:
`this.$route.matched` is an array that contains [all matched routes in parent-to-child order](https://v3.router.vuejs.org/api/#route-object-properties). E.g. when visiting url `manager/installed`, it will contain `RouteRecord`s for both the route named "manager" and the one named "manager.installed". Currently we inspect only `this.$route.matched[0]`, i.e. the main level route determines whether a sidebar is shown for all/none of its child routes. So for main level routes like `SettingsView` or `Help` that don't have subroutes, it just check if those routes themself define `NavigationMenu` component.

If we come up with a situation where a subroute needs a navigation menu while the parent doesn't, I think we can change this to check routes starting from the last child and moving backwards. But for now, since there's no such use case, I think it's nice to have to define the navigation component only once in the parent route.

Reply:
Marking this resolved since the code in question is no longer present in PR.